src/components/App/App.js

```diff
@@ -8,12 +8,64 @@ import { SavedNews } from "../SavedNews/SavedNews";
 import { PopUpRegister } from "../PopUpRegister/PopUpRegister";
 import { PopUpLogin } from "../PopUpLogin/PopUpLogin";
 import { PopupWithForm } from "../PopupWithForm/PopupWithForm";
+import newsApi from "../../utils/NewsApi";
+import mainApi from "../../utils/MainApi";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
+import { UserSavedCardsContext } from "../../contexts/UserSavedCardsContext";
 
 function App() {
   const [isLogedIn, setIsLogedIn] = useState(false);
   const [registerPopUpOpen, setRegisterPopUpOpen] = useState(false);
   const [loginPopUpOpen, setLoginPopUpOpen] = useState(false);
   const [registretionSuccsess, setRegistretionSuccsess] = useState(false);
+  const [newsCardListStatus, setNewsCardListStatus] = useState("");
+  const [newsCards, setNewsCards] = useState([]);
+  const [userCard, setUserCard] = useState(
+    JSON.parse(localStorage.getItem("userCards"))
+  );
+  const [registerErrorMsg, setRegisterErrorMsg] = useState("");
+  const [token, setToken] = useState(localStorage.getItem("jwt"));
+  const [currentUser, setCurrentUser] = useState({});
+  const searchKeyWord = localStorage.getItem("searchKeyWord");
+
+  function ProtectedRoute({ children, isLogedIn }) {
+    return !isLogedIn ? <Navigate replace to="/" /> : children;
+  }
+
+  React.useEffect(() => {
+    if (!localStorage.getItem("searchResults")) {
+      return;
+    }
+    const localStorageCards = JSON.parse(localStorage.getItem("searchResults"));
+    setNewsCards(localStorageCards);
+    setNewsCardListStatus("Finish");
+  }, []);
+
+  React.useEffect(() => {
+    if (!localStorage.getItem("jwt")) {
+      return;
+    }
+    mainApi.getUserInfo(token).then((res) => {
+      setIsLogedIn(true);
+      setCurrentUser(res);
+    });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

Please, correct it everywhere